### ISSUE TITLE
Add support for String-valued EJSON-wrapper for `decimal` for input

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/data-api#1654]
(note: follow-up to #1608, discussions)

Due to challenges some clients (in particular Python and/or Typescript) have in encoding JSON number from values other than binary 32-bit and/or 64-bit floating-point values (`float` / `double`), it seems necessary to allow alternate representation to be sent: something that you would like

```json
{ "$decimal": "0.001" }
```

and possibly for convenience also redundant:

```json
{ "$decimal": 0.001 }
```

Return value will still use "simple" (aka "relaxed") plain JSON number representation

```json
0.001
```

Note that is specifically limited to CQL `decimal` type, at this point.

**Comment by amorton** [comment]
on hold until we find a good use case for this.

**Comment by vkarpov15** [comment]
I'm not sure I understand what issues the clients have in encoding JSON numbers? #1608 is unclear about what the issues are. Granted if you try to send the result of `0.1 + 0.2` from a Node.js application in JSON, you'll send `0.30000000000000004`, but that would be the case regardless of whether you serialize the result to a string, number, or `{ $decimal }` wrapper.

**Comment by tatu-at-datastax** [comment]
@vkarpov15 I agree with your points and to me this does not seem like something to implement. But as per description, I created this to allow client team to argue for implementation, explain the issue(s) in more detail in case I misunderstood or missed something.

**Comment by vkarpov15** [comment]
Just wanted to follow up since I've gotten some more clarity on this issue, more info [here](https://github.com/stargate/data-api/issues/1608#issuecomment-2567828938):

The issue is that built-in JSON parsers in some languages, like JavaScript, are limited by built-in number limits. For example, if you try to parse `9223372036854775807` using `JSON.parse()` in JavaScript, you lose precision.

```
$ node
Welcome to Node.js v20.10.0.
Type ".help" for more information.
> const str = '9,223,372,036,854,775,807'.replace(/,/g, '')
undefined
> str
'9223372036854775807'
> JSON.parse(`{"value":${str}}`)
{ value: 9223372036854776000 }
> `{"value":${str}}`
'{"value":9223372036854775807}'
> 
```

As a client of Data API, my options are to either 1) use a custom JSON parser, which may be slower and less fully featured than the native parser; 2) try to do some custom parsing to detect large numbers before passing to JSON.parse(); 3) ignore precision errors and potentially return incorrect values to users.
